api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java

```diff
@@ -63,11 +65,27 @@ public ClaimResult createOrUpdateClaim(Claimant claimant) {
 
             return createResult(claim);
         } catch (RuntimeException e) {
-            createAndSaveClaim(claimant, buildWithStatus(EligibilityStatus.ERROR));
+            handleFailedClaim(claimant, e);
             throw e;
         }
     }
 
+    private void handleFailedClaim(Claimant claimant, RuntimeException e) {
+        Claim claim = buildClaim(claimant, buildWithStatus(EligibilityStatus.ERROR));
+        NewClaimEvent newClaimEvent = NewClaimEvent.builder()
+                .claimId(claim.getId())
+                .claimStatus(claim.getClaimStatus())
+                .eligibilityStatus(claim.getEligibilityStatus())
+                .build();
+        FailureEvent failureEvent = FailureEvent.builder()
+                .failureDescription("Unable to create (or update) claim")
+                .failedEvent(newClaimEvent)
+                .exception(e)
+                .build();
+        eventAuditor.auditFailedEvent(failureEvent);
```

Comment:
Corrected to audit a failed event instead of a new claim event, and ensured the audit happens before the save (as there's a good chance at this point that the failure to save the new claim was due to db errors).